scripts/protoc.sh

```diff
@@ -9,44 +9,47 @@
 set -e
 
 REPO=github.com/tinkerbell/pbnj
-REPO_BASE=$(dirname ${REPO})
 PROTOS_LOC=api/v1
-OUT_LOC=api/v1
+PROTOC_VERSION=3.13.0
 
 function installDeps {
     if ! which protoc 2>&1 >/dev/null; then
         echo 'Installing protoc...' >&2
-        if which apt 2>&1 >/dev/null; then
-            apt update; apt install -y protobuf-compiler
-        elif which yum 2>&1 >/dev/null; then
-            yum -y install protobuf-compiler
-        else
-            echo 'Unknown package manager' >&2
-            exit 1
+        if ! which unzip 2>&1 >/dev/null; then
+            if which apt 2>&1 >/dev/null; then
+                apt update; apt install -y zip
+            elif which yum 2>&1 >/dev/null; then
+                yum -y install zip
+            else
+                echo 'Unknown package manager' >&2
+                exit 1
+            fi
         fi
+        PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+        curl -LO $PB_REL/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip
+        unzip protoc-${PROTOC_VERSION}-linux-x86_64.zip -d /usr/local
         echo 'Protoc installed' >&2
     else
         echo 'Protoc already installed!' >&2
     fi
     go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.0.1
     go get google.golang.org/grpc@v1.32.0
     go get google.golang.org/protobuf/cmd/protoc-gen-go@v1.25.0
+    go get github.com/mwitkow/go-proto-validators/protoc-gen-govalidators@v0.3.2
     go get golang.org/x/tools/cmd/goimports
-    go mod tidy
+    go mod tidy || true
 }
 
 if [[ "$1" == "deps" ]]; then
     installDeps
     exit 0
 fi
+pbs=$(find ${PROTOS_LOC} -type f -name '*.proto'| xargs | tr " " ",")
 
-mkdir -p ${REPO_BASE}
-ln -nsf ${PWD} ${REPO_BASE}
-for elem in $(find ${PROTOS_LOC} -type f -name '*.proto'); do
-    echo -e "Generating .pb.go for ${elem}"
-    protoc -I . -I github.com --go_out=${PWD} --go_opt=module=${REPO} ${elem} || true
-    protoc -I . -I github.com --go-grpc_out=${PWD} --go-grpc_opt=module=${REPO} ${elem} || true
-done
-rm -f ${REPO}
-rmdir -p ${REPO_BASE}
+echo -n "Generating code from protocol buffers (${pbs})..."
+protoc -I . -I $(go env GOMODCACHE) --go_out=. --go_opt=module=${REPO} ${PROTOS_LOC}/*.proto
+protoc -I . -I $(go env GOMODCACHE) --govalidators_out=. --go-grpc_out=. --go-grpc_opt=module=${REPO} ${PROTOS_LOC}/*.proto
+mv ${REPO}/${PROTOS_LOC}/*.go ${PROTOS_LOC}/
+rm -rf github.com
 $(go env GOPATH)/bin/goimports -w . || true
+echo -ne "done\n"
```

Comment:
Why not just `echo "done"` ?

Reply:
this just makes the output something like this 
```bash
Generating code from protocol buffers (${pbs})...done
```
instead of 
```
Generating code from protocol buffers (${pbs})...
done
```

Reply:
No the result would actually be the same, as long as you leave line 49 alone, as that is the line you're telling it to not return at.

Reply:
But this isn't an actual issue, just informative

Reply:
ahh sorry, I see it now. your bash-fu is much stronger than mine.

Reply:
updated